plutus-scb/src/Plutus/SCB/Events.hs

```diff
@@ -9,31 +11,35 @@ module Plutus.SCB.Events
     , module Events.Node
     , module Events.Wallet
     , ChainEvent(..)
-    , _RecordRequest
-    , _RecordResponse
     , _UserEvent
     , _NodeEvent
     , _WalletEvent
+    , _ContractEvent
     ) where
 
 import           Control.Lens.TH
 import           Data.Aeson                 (FromJSON, ToJSON)
+import           Data.Text.Prettyprint.Doc
 import           GHC.Generics               (Generic)
 import           Plutus.SCB.Events.Contract as Events.Contract
 import           Plutus.SCB.Events.Node     as Events.Node
 import           Plutus.SCB.Events.User     as Events.User
 import           Plutus.SCB.Events.Wallet   as Events.Wallet
 
 -- | A structure which ties together all possible event types into one parent.
-data ChainEvent t
-    = RecordRequest
-          !(Events.Contract.RequestEvent Events.Contract.ContractRequest)
-    | RecordResponse
-          !(Events.Contract.ResponseEvent Events.Contract.ContractResponse)
-    | UserEvent !(Events.User.UserEvent t)
+data ChainEvent t =
+    UserEvent !(Events.User.UserEvent t)
     | NodeEvent !Events.Node.NodeEvent
     | WalletEvent !Events.Wallet.WalletEvent
+    | ContractEvent !(Events.Contract.ContractEvent t)
     deriving (Show, Eq, Generic)
     deriving anyclass (FromJSON, ToJSON)
 
 makePrisms ''ChainEvent
+
+instance Pretty t => Pretty (ChainEvent t) where
+    pretty = \case
```

Comment:
I feel like we should have a deriving class for this somehow :thinking: not bothered enough to write one, though

Reply:
One day...